suite-common/wallet-utils/src/cardanoStakingUtils.ts

```diff
@@ -31,12 +32,56 @@ export const getCardanoStakingSymbols = (networkSymbols: NetworkSymbol[]) =>
         return acc;
     }, [] as SupportedCardanoNetworkSymbols[]);
 
-export const isCardanoStakedWithEverstake = (account: Account) => {
-    if (!account) return false;
+export const isCardanoStakedWithEverstake = (
+    account: Account,
+    cardanoStakingPools: CardanoPoolInfo[],
+) => {
+    if (!account || !cardanoStakingPools?.length) return false;
 
     if (account.networkType !== 'cardano') return false;
 
-    return account?.misc.staking.poolId === CARDANO_EVERSTAKE_STAKING_POOL.bech32;
+    const accountPoolId = account.misc?.staking?.poolId;
+    if (!accountPoolId) return false;
+
+    return cardanoStakingPools.some(pool => pool.id === accountPoolId);
+};
+
+export const poolBech32ToHex = (poolId: string): string => {
+    const decoded = bech32.decode(poolId);
+    const bytes = bech32.fromWords(decoded.words);
+
+    return Buffer.from(bytes).toString('hex');
+};
+
+export const selectBestCardanoPool = (pools?: CardanoPoolInfo[]) => {
+    if (!pools || pools.length === 0) return;
+
+    // pools that are under the safe saturation threshold
+    const safePools = pools.filter(
+        pool => pool.saturation < CARDANO_POOL_SATURATION_SAFE_THRESHOLD,
+    );
+
+    if (safePools.length > 0) {
+        // pick the safe pool with the highest saturation
+        const bestPool = safePools.reduce((best, curr) =>
+            curr.saturation > best.saturation ? curr : best,
+        );
+
+        return {
+            hex: poolBech32ToHex(bestPool.id),
+            bech32: bestPool.id,
+        };
+    }
+
+    // otherwise, choose the pool with the lowest saturation overall
+    const lowestSaturationPool = pools.reduce((best, curr) =>
+        curr.saturation < best.saturation ? curr : best,
+    );
+
+    return {
```

Comment:
```suggestion
    // sort from highest saturation to lowest
    const sortedPools = [...pools].sort((a, b) => b.saturation - a.saturation);
    
    // find the one within the threshold
    const bestPool = sortedPools.find(pool => pool.saturation < CARDANO_POOL_SATURATION_SAFE_THRESHOLD)
    
    if (bestPool) {
      return {
            hex: poolBech32ToHex(bestPool.id),
            bech32: bestPool.id,
        };
    }
    
    // pick the last one (lowest saturation)
    const fallback = sortedPools[sortedPools.length - 1];
    return {
        hex: poolBech32ToHex(fallback.id),
        bech32: fallback.id,
    };
```

I believe this approach is cleaner than using `reduce`

Reply:
what can I say, that's my inner reduce-lover showing up again 😅 Fixed here https://github.com/trezor/trezor-suite/commit/fcbcec3f179da3748c28ecbb596b80a6f4508572